core/src/main/java/cz/xtf/core/openshift/OpenShiftWaiters.java

```diff
@@ -120,16 +125,44 @@ public Waiter isProjectReady() {
     public Waiter isProjectClean() {
         return new SimpleWaiter(() -> {
             int crdInstances = 0;
+            List<GenericKubernetesResource> customResourceDefinitionList = null;
             for (CustomResourceDefinitionContextProvider crdContextProvider : OpenShift.getCRDContextProviders()) {
                 try {
-                    crdInstances += ((List) (openShift.customResource(crdContextProvider.getContext())
-                            .list(openShift.getNamespace()).get("items"))).size();
+                    customResourceDefinitionList = openShift.customResources(crdContextProvider.getContext(),
+                            GenericKubernetesResource.class, GenericKubernetesResourceList.class)
+                            .inNamespace(openShift.getNamespace())
+                            .list().getItems();
+                    crdInstances += customResourceDefinitionList.size();
                 } catch (KubernetesClientException kce) {
                     // CRD might not be installed on the cluster
                 }
             }
-            return crdInstances == 0 & openShift.listRemovableResources().isEmpty();
-        }, TimeUnit.MILLISECONDS, WaitingConfig.timeoutCleanup(), "Cleaning project.")
+
```

Comment:
I take this is not directly related to  `xtf.openshift.namespace.per.testcase` changes? But just some troubelshooting logging, right?